Shared/sdk/CScriptArgReader.h

```diff
@@ -1052,6 +1071,31 @@ class CScriptArgReader
     }
 
 protected:
+    void InternalReadPair(std::vector<std::pair<SString, SString>>& outPairs, int iIndex)
+    {
+        lua_pushnil(m_luaVM);
+        while (lua_next(m_luaVM, iIndex) != 0)
+        {
+            int keyType = lua_type(m_luaVM, -2);
+            int valueType = lua_type(m_luaVM, -1);
+            if (keyType == LUA_TSTRING)
+            {
+                SString value;
+                if (valueType == LUA_TSTRING || valueType == LUA_TNUMBER)
+                {
+                    uint uiLength = lua_strlen(m_luaVM, -1);
+                    value.assign(lua_tostring(m_luaVM, -1), uiLength);
+                }
+                else if (valueType == LUA_TBOOLEAN)
+                {
+                    value = (lua_toboolean(m_luaVM, -1) ? "1" : "0");
+                }
```

Comment:
Feel like a switch case would be more readable?

Reply:
It's more a matter of taste. I personally think a pure branching in this case is more readable.